sparkmagic/sparkmagic/controllerwidget/addendpointwidget.py

```diff
@@ -59,3 +61,12 @@ def run(self):
         # We need to call the refresh method because drop down in Tab 2 for endpoints wouldn't refresh with the new
         # value otherwise.
         self.refresh_method()
+
+    def update_endpoint(self):
```

Comment:
I believe adding old_value, new_value are optional and the values can be accessed in the callback as well.